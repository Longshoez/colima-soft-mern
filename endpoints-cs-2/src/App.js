import React from 'react';
import { GifExpertApp } from './Components/GifExpertApp/GifExpertApp';

export const App = () => {
  return(
      <GifExpertApp />
  );
};
