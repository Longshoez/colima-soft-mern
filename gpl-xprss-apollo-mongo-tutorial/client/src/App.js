import './App.css';
import { useMutation, useQuery} from '@apollo/client'
import { getAll } from './GraphQl/Query';
import { CREATE_POST, DELETE_POST } from './GraphQl/Mutation';
import { useState } from 'react';
import { Task } from './Task';

function App() {
  const {loading, error, data} = useQuery(getAll)

  const [inputTitle, setTitle] = useState('')
  const [inputDescription, setDescription] = useState('')


  //here we asign an identifier for the mutation function, instead of using CREATE_POST we use createPost, but we have the logic from CREATE_POST
  const [createPost, {err}] = useMutation(CREATE_POST)
  const [deletePost, {errr}] = useMutation(DELETE_POST)

  if (loading) return 'Loading'
  if (error)return 'Error'  
  if (data) {
    console.log(data)
  }

  const addPost = () => {
    createPost({
      variables:{
        title: inputTitle,
        description: inputDescription
      }
    })
  }

  const removePost = (id) => {
    deletePost({
      variables:{
        id: id,
      }
    })
  }

  return (  
    <div className="App">      
      {
        data.getAll.map((data, index) => (          
          <Task index={index} title={data.title} description={data.description} func={() => removePost(data.id)}/>
        ))
      }              
      <div className='nav'>
        <div>
          <input type="text" onChange={(e) => setTitle(e.target.value)} placeholder="New task"/>
          <textarea type="text" onChange={(e) => setDescription(e.target.value)} placeholder="Task description"/>
        </div>
        <button onClick={() => addPost()}>Add post</button>
      </div>  
  </div>
  );
}

export default App;
