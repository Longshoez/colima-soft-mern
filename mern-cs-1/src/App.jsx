import './App.css'
import {DatePicker, Card} from 'antd'
import 'antd/dist/antd.css'


function App() {

  const test = (date, dateString) => {
    console.log(date, dateString)
  }
  

  return (
    <div className="App">
      yeah
      <Card 
        title="yeahleah bleah"
        extra={<a href="#"> More</a>}
        style={{width: 300}}
      >
        <p>card content</p>
        <p>card content</p>
        <p>card content</p>
      </Card>

      <DatePicker onChange={test}/>
    </div>
  )
}

export default App
