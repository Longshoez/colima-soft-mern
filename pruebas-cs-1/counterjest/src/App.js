import { useState } from 'react';
import './App.css';

function App({value = 10}) {

  const [counter, setCounter] = useState(value);

  return (
    <div className="App">
      <header className="App-header">
        <button type="button" onClick={() => setCounter(() => counter+1)}>+1</button>
        <h6 onClick={() => setCounter((value))}>{counter}</h6>
        <button type="button" onClick={() => setCounter(() => counter-1)}>-1</button>
      </header>
    </div>
  );
}

export default App;