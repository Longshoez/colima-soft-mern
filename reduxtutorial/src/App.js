import './App.css';
import 'bootstrap/dist/css/bootstrap.min.css';
//0import CantidadPokemon from './components/CantidadPokemon';
//import CompraPokemon from './components/CompraPokemon';

import {CantidadPokemon} from './components/CantidadPokemon.hook'
import {ComprarPokemon} from './components/ComprarPokemon.hook'

import BuscadorPokemon from './components/BuscadorPokemon'
import ResultadoPokemon from './components/ResultadoPokemon';

import store from './redux/store'
import { Provider } from 'react-redux'

function App() {
  return (
    <Provider store={store}>
    <div className="App container">
      <div className="row">
        <div className="col-23">
          <div className="card mt-t" style={{maxWidth: '370px'}}></div>
          <div className="row no-gutters ">
            <div className="col-4">
              <img src={require('./golden.jpeg')} alt="its a golden" className='img-thumbnail'/>
            </div>
            <div className="col-8">
              <div className="card-body">
                <CantidadPokemon />   
                <ComprarPokemon />
              </div>
            </div>
          </div>
        </div>
        <div className="col-12 mt-4 pt-3">
          <BuscadorPokemon/>
        </div>
        <div className="col-12">
          <ResultadoPokemon />
        </div>
      </div>
    </div>
    </Provider>
  );
}

export default App;
