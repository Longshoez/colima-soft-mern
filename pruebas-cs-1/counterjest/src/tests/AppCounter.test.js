import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';


describe('Testeando el counter en el componente App', () => {

    //We declare our component here so we can use it multiple times through the test withouot needing to import it each time
    let wrapper = shallow(<App/>)

    //Reset the component
    beforeEach(() => {
        wrapper = shallow(<App/>)
    })

    test('should first show the counter app and its default value', () => {
        
        //Arrrange
        const wrapper = shallow(
            <App/>
        )
        //Act
        //Assert
        expect(wrapper).toMatchSnapshot();
        
    });

    test('should show 100 as a default value', () => {
        //Arrrange
        const value = 100;
        const wrapper = shallow(
            <App value={value}/>
        )

        //Act
        const Counter = wrapper.find('h6').text().trim() //trim remvoes the spaces

        //Assert
        expect(Counter).toBe('100');
    });

    test('should increase when clicking the +1 button', () => {
        //Arrrange
        //Wrapper with componoent defined in the upper lvl scope

        
        //Act
        const button1 = wrapper.find('button').at(0).simulate('click')
        console.log(button1.html());
        
        //Assert
        const Counter = wrapper.find('h6').text().trim()
        expect(Counter).toBe('11')

    });

    test('should decrease when clicking the +1 button', () => {
        //Arrrange
        //Wrapper with componoent defined in the upper lvl scope

        //Act
        const button2 = wrapper.find('button').at(1).simulate('click')
        console.log(button2.html());

        
        //Assert
        const Counter = wrapper.find('h6').text().trim();
        expect(Counter).toBe('9')

    });

    test('should reset the counter to its default value', () => {
        //Arrrange
        const value = 105;

        const wrapper = shallow(
            <App value={value}/>
        )
        
        //Act   
        const resetButton = wrapper.find('h6').simulate('click')
        console.log(resetButton.html());

        
        //Assert
        const Counter = wrapper.find('h6').text().trim() //trim remvoes the spaces
        expect(Counter).toBe(`${value}`);
    });
    

    
})