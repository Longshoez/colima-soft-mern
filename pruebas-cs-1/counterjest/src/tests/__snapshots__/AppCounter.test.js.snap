// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testeando el counter en el componente App should first show the counter app and its default value 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      +1
    </button>
    <h6
      onClick={[Function]}
    >
      10
    </h6>
    <button
      onClick={[Function]}
      type="button"
    >
      -1
    </button>
  </header>
</div>
`;
